### PR TITLE
Filter by property

### PR DESCRIPTION
**Pull request by dev4ndy**
[Rally](https://rally1.rallydev.com/#/340101526180d/detail/userstory/370157356516)

### CODE REVIEW

File: src/cardstack/al-cardstack-view.ts [hunk 1]
```diff
@@ -373,25 +373,41 @@ export abstract class AlCardstackView< EntityType=any,
             if (this.characteristics.searchableBy.length === 0 ) {
                 return true;
             }
-            this.characteristics.searchableBy.find( (property:string) => {
-                if ( ! card.properties.hasOwnProperty( property ) || typeof( ( card.properties as any)[property] ) === 'undefined' ) {
-                    return false; //  terminate iteration
+            if(search.source.indexOf("=") !== -1){
```
Comment (scottarver) [inline src/cardstack/al-cardstack-view.ts]: use .includes

File: src/cardstack/al-cardstack-view.ts [hunk 2]
```diff
@@ -373,25 +373,41 @@ export abstract class AlCardstackView< EntityType=any,
...
-                if (cardPropValue instanceof Array) {
-                    const matches = cardPropValue.find((value) => search.test(value));
-                    if (matches) {
-                        visible = true;
-                        return true;
+                if(!card.properties.hasOwnProperty(property)){
+                    return false;
+                }
+                const cardPropValue = ( card.properties as any )[property].toLowerCase();
+                if (value.indexOf(cardPropValue) !== -1) {
```
Comment (scottarver) [inline src/cardstack/al-cardstack-view.ts]: use .includes

File: src/cardstack/al-cardstack-view.ts [hunk 3]
```diff
@@ -373,25 +373,41 @@ export abstract class AlCardstackView< EntityType=any,
...
+                    if ( ! card.properties.hasOwnProperty( property ) || !( card.properties as any)[property]) {
+                        return true; // terminate iteration
                     }
-                } else {
-                    if (search.test(cardPropValue)) {
-                        visible = true;
-                        return true;
+                    let cardPropValue = ( card.properties as any )[property];
+                    if (cardPropValue instanceof Array) {
+                        const matches = cardPropValue.find((value) => search.test(value));
```
Comment (scottarver) [inline src/cardstack/al-cardstack-view.ts]: use `.some`

File: src/cardstack/al-cardstack-view.ts (hunk 1 above)
Comment (dev4ndy) [reply src/cardstack/al-cardstack-view.ts]: Includes is not supporting on IE11

Comment (mcnielsen) [reply src/cardstack/al-cardstack-view.ts]: God bless polyfills...  definitely prefer `.includes()` for clarity.

**Pull request merged**